tcmur_cmd_handler.c

```diff
@@ -972,102 +974,118 @@ static int xcopy_parse_parameter_list(struct tcmu_device *dev,
 	return ret;
 }
 
-static void handle_xcopy_rw_sync(struct tcmu_device *dev,
-				 struct tcmulib_cmd *cmd,
-				 int ret)
+static int xcopy_read_work_fn(struct tcmu_device *src_dev, struct tcmulib_cmd *cmd);
+static void handle_xcopy_read_cbk(struct tcmu_device *src_dev,
+				  struct tcmulib_cmd *cmd,
+				  int ret);
+
+static void handle_xcopy_write_cbk(struct tcmu_device *dst_dev,
+				  struct tcmulib_cmd *cmd,
+				  int ret)
 {
 	struct xcopy *xcopy = cmd->cmdstate;
+	struct tcmu_device *src_dev = xcopy->src_dev;
+
+	/* write failed - bail out */
+	if (ret != SAM_STAT_GOOD) {
+		tcmu_dev_err(src_dev, "Failed to write to dst device!\n");
+		ret = tcmu_set_sense_data(cmd->sense_buf, MEDIUM_ERROR,
+					  ASC_WRITE_ERROR, NULL);
```

Comment:
Here and below there is not need to set the sense. The handler will have set it based on whatever error it hit.